helix-core/src/selection.rs

```diff
@@ -635,7 +630,20 @@ impl Selection {
                     };
                 }
                 (None, Some(_)) => {
-                    // exhausted `self`, we can't match the reminder of `other`
+                    // Other might still contain all of its ranges in a single range of self,
+                    let mut still_subset: bool;
+                    for self_range in self {
```

Comment:
So I think there are several issues here:

1. This is not necessary, as mentioned above
2. It actually makes it less correct, because you could have multiple cross sections of ranges where a larger range in the first overlaps with multiple smaller ranges in the second, e.g.
    
    ```
    [a b c d e f g]  [h i j k l m n]
    [a]b c[d]e f[g]  [h]i j[k]l m[n]
    ```
    
    In this case, it will say it doesn't contain it because of `hkn`
3. It is currently an incremental algorithm linear on the number of ranges in the selections, but this makes it exponential.

Reply:
You're right. I had overlooked exactly what was being done in the first match arm. I also expanded the test case for the example case class you gave. Should be resolved now, thank you.